src/AspNet.Security.OAuth.Baidu/BaiduAuthenticationOptions.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http;
+
+namespace AspNet.Security.OAuth.Baidu
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="BaiduAuthenticationHandler"/>.
+    /// </summary>
+    public class BaiduAuthenticationOptions : OAuthOptions
+    {
+        public BaiduAuthenticationOptions()
+        {
+            ClaimsIssuer = BaiduAuthenticationDefaults.Issuer;
+            CallbackPath = new PathString(BaiduAuthenticationDefaults.CallbackPath);
+
+            AuthorizationEndpoint = BaiduAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = BaiduAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = BaiduAuthenticationDefaults.UserInformationEndpoint;
+            
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "uid");
+            ClaimActions.MapJsonKey(ClaimTypes.Name, "uname");
+            ClaimActions.MapCustomJson(BaiduAuthenticationConstants.Claims.Portrait,
+                user => $"https://tb.himg.baidu.com/sys/portrait/item/{user.Value<string>("portrait")}");
```

Comment:
If there wasn't a value for this in the response it would produce a malformed claim.

Reply:
the `portrait` is a hash (or something else) only for part of avatar picture url.

Reply:
But if it wasn't returned in the response for some reason you'd just end up with `https://tb.himg.baidu.com/sys/portrait/item/`

Reply:
u r right